### PR TITLE
Refactoring combat hooks

### PR DESCRIPTION
**Pull request by FakelsHub**
Optional correction for displaying the correct damage during instant death.

Ammocost hook changes:
Fixed arg2 / arg3 arguments for hook type 1/2 - now in the mode of a burst mode, the number of rounds in the burst is returned.
The arg4 argument has an additional type 3.
```
0 - when subtracting ammo after single shot.
1 - when checking for "out of ammo" before attack (req. CheckWeaponAmmoCost)
2 - correction for the calculation of ammo for burst rounds (req. CheckWeaponAmmoCost)
3 - when subtracting ammo after burst attack
```
and for hook type 1 (CheckWeaponAmmoCost option) added a check when burst attack.

### CODE REVIEW

File: sfall/Modules/BugFixes.cpp [hunk 1]
```diff
@@ -1112,18 +1113,55 @@ static void __declspec(naked) obj_move_to_tile_hack() {
...
+		add  esp, 0x34;
+		pop  ebp;
+		pop  edi;
+		jmp  ComputeDamageRet;
+	}
+}
 
 void BugFixes::init()
 {
+	if (isDebug && (GetConfigInt("Debugging", "BugFixes", 1) == 0)) return;
```
Comment (NovaRain) [inline sfall/Modules/BugFixes.cpp]: This is rather redundant IMO, along with the InstantDeathTypeFix below.

**Comment by FakelsHub** [comment]
> BugFixes

I need this for testing because sfall inserts of the edits prevent from analyzing the code.
This parameter need not be included in the ini - the more it should not interfere with you.

> InstantDeathTypeFix

Which variant of the fix to use the old one or the new one? or what you offer.

**Comment by NovaRain** [comment]
>I need this for testing because sfall inserts of the edits prevent from analyzing the code.
This parameter need not be included in the ini - the it should not interfere with you.

Then I'd say you should just keep it in your own test build, that's what I did when testing/verifying stuff in the past. I don't like the "hidden settings" idea in some old sfall builds.

>Which variant of the fix to use the old one or the new one? or what you offer.

I'd prefer the old one, as the new one only changes to display target's full health for insta-death crit. Not really much an functional improvement other than satisfying some people's feel when reading messages if you ask me.

**Comment by FakelsHub** [comment]
> I'm not fond of the "hidden settings" idea in some old sfall builds.

I saved you from "hidden settings" :)

**Pull request merged**